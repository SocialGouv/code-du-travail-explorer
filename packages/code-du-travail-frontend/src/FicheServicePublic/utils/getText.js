export default function getText(element = { text: "" }, joint = " ") {
  if (element.type === "text") {
    return element.$.trim();
  }
  if (element.$) {
    return element.$.map(child => getText(child, joint)).join(joint);
  }
  return "";
}
