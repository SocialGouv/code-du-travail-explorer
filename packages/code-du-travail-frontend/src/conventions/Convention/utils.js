import { matopush } from "../../piwik";

export function trackAccordionPanelState(conventionName, eventName) {
  let previousPanelState = [];
  return function (currentPanelState) {
    if (currentPanelState.length > previousPanelState.length) {
      const lastOpenedPanel = currentPanelState.filter(
        (panel) => !previousPanelState.includes(panel)
      );
      matopush(["trackEvent", eventName, conventionName, lastOpenedPanel[0]]);
    }
    previousPanelState = currentPanelState;
  };
}
