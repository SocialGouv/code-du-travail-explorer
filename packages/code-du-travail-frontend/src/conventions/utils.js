export const sortByIntOrdre = (a, b) => {
  if (a.intOrdre < b.intOrdre) {
    return -1;
  }
  if (a.intOrdre > b.intOrdre) {
    return 1;
  }
  return 0;
};
