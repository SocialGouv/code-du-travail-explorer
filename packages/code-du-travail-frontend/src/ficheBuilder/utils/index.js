export { default as ignoreParagraph } from "./ignoreParagraph";
export { default as getText } from "./getText";
