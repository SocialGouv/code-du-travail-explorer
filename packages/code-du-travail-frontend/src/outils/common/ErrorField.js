import { theme } from "@socialgouv/react-ui";
import PropTypes from "prop-types";
import React from "react";
import { Field } from "react-final-form";
import styled from "styled-components";

function ErrorField({ name }) {
  return (
    <Field
      name={name}
      subscription={{ error: true, submitFailed: true, touched: true }}
      render={({ meta: { error, submitFailed } }) => {
        if (!submitFailed || !error) {
          return null;
        }
        if (typeof error === "string") {
          return <Error>{error}</Error>;
        }
        return error;
      }}
    />
  );
}
ErrorField.propTypes = {
  name: PropTypes.string.isRequired,
};

function ErrorComputedField({ name }) {
  return (
    <Field
      name={name}
      render={({ meta: { error } }) => (error ? <Error>{error}</Error> : null)}
    />
  );
}
ErrorComputedField.propTypes = {
  name: PropTypes.string.isRequired,
};

const Error = styled.div`
  margin: ${theme.spacings.small} 0 ${theme.spacings.base};
  color: ${({ theme }) => theme.error};
  font-weight: 600;
  font-size: ${theme.fonts.sizes.small};
`;

const InlineError = (props) => <Error {...props} as="span" />;

export { ErrorField, ErrorComputedField, Error, InlineError };
