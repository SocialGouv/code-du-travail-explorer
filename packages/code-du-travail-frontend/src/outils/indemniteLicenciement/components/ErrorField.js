import React from "react";
import PropTypes from "prop-types";
import { Field } from "react-final-form";
import { Alert } from "@cdt/ui";

function ErrorField({ name, immediate = false }) {
  return (
    <Field
      name={name}
      subscribe={{ error: true, touched: true, visited: immediate }}
      render={({ meta: { touched, error, visited } }) =>
        (error && touched) || (error && visited) ? <Alert>{error}</Alert> : null
      }
    />
  );
}

ErrorField.propTypes = {
  name: PropTypes.string.isRequired,
  immediate: PropTypes.bool
};

export { ErrorField };
