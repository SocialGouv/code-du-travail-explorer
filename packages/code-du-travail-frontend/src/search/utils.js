import { getRouteBySource, SOURCES } from "@cdt/sources";

import { matopush } from "../piwik";

export const summarize = (text = "") =>
  text.length > 160 ? text.slice(0, text.indexOf(" ", 160)) + "â€¦" : text;

export const formatUrlMatomo = (source, slug, url) => {
  const trackedUrl =
    source === SOURCES.EXTERNALS ? url : `/${getRouteBySource(source)}/${slug}`;

  return trackedUrl;
};

export const reportSelectionToMatomo = (source, slug, url, algo) => {
  const qualifiedCall = JSON.stringify({
    algo,
    url: formatUrlMatomo(source, slug, url),
  });

  matopush(["trackEvent", "selectResult", qualifiedCall]);
};
