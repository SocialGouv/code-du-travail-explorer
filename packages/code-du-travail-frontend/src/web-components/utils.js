export const displayInViewport = (target, rootBoundaries) => {
  const viewportHeight =
    window.innerHeight || document.documentElement.clientHeight;
  const viewportWidth =
    window.innerWidth || document.documentElement.clientWidth;

  target.style.display = "block";
  target.style.top = rootBoundaries.height + "px";

  let targetBounding = target.getBoundingClientRect();

  target.style.left =
    -Math.round((targetBounding.width - rootBoundaries.width) / 2) + "px";

  targetBounding = target.getBoundingClientRect();
  // we are overflowing on the right
  if (targetBounding.right > viewportWidth) {
    target.style.right = "0px";
    target.style.left = "auto";
  }
  // we are overflowing on the left
  if (targetBounding.left < 0) {
    target.style.left = "0px";
    target.style.right = "auto";
  }
  // we are overflwing to the bottom
  if (targetBounding.bottom > viewportHeight) {
    target.style.top = -Math.round(targetBounding.height) + "px"; // don't make it go below header
  }
  // second pass to make sure the first pass did not
  // make the tooltip overflow on the opposite side :o
  targetBounding = target.getBoundingClientRect();
  if (targetBounding.left < 0) {
    target.style.right = targetBounding.left - 10 + "px";
  }
  if (targetBounding.right > viewportWidth) {
    target.style.left = viewportWidth - targetBounding.right - 10 + "px";
  }
  // don't make it go below header
  if (target.getBoundingClientRect().top < 120) {
    target.style.top = rootBoundaries.height + "px";
  }
};

const throttler = function (fn, interval) {
  let ready = true;
  return function (...args) {
    if (!ready) return;
    if (ready) {
      ready = false;
      window.setTimeout(() => {
        ready = true;
      }, interval);
      return fn(...args);
    }
  };
};

export const throttledDisplayInViewport = throttler(displayInViewport, 10);
