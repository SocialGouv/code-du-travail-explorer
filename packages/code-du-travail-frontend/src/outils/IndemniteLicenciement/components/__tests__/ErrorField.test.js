import React from "react";
import { render, fireEvent } from "../../../../../test/utils";
import { ErrorField } from "../ErrorField";
import { Form, Field } from "react-final-form";

describe("<ErroField />", () => {
  it("should render error", () => {
    const onSubmit = jest.fn();
    const { container, getByTestId } = render(
      <Form
        initialValues={{ test: "a" }}
        onSubmit={onSubmit}
        render={() => (
          <>
            <Field
              data-testid="test"
              name="test"
              component="input"
              validate={value => (value ? undefined : "Champ requis")}
            />
            <ErrorField name="test" />
          </>
        )}
      />
    );
    const input = getByTestId("test");
    fireEvent.change(input, { target: { value: "" } });
    input.blur();
    expect(container).toMatchSnapshot();
  });
});
