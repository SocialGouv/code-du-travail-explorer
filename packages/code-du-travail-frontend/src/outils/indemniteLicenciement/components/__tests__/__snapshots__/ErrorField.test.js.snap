// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErroField /> should render error 1`] = `
<div>
  <input
    data-testid="test"
    name="test"
    value=""
  />
  <div
    class="alert "
  >
    Champ requis
  </div>
</div>
`;
